primap2/_convert.py

```diff
@@ -12,12 +12,13 @@
 
 
 class DataArrayConversionAccessor(_accessor_base.BaseDataArrayAccessor):
-    @alias_dims(["dim"])
-    def convert(
+    def _convert_inner(
         self,
         dim: Hashable | str,
-        categorization: climate_categories.Categorization | str,
         *,
+        # TODO type will change to climate_categories.Conversion when
+        #  https://github.com/primap-community/climate_categories/pull/164 is merged
+        conversion: climate_categories._conversions.Conversion,
```

Comment:
Now that I see the definition of `_convert_inner`, I think we can go all-in with this approach. As you said in chat, we can leave the whole "get the right conversion" business to `climate_categories`, where we have functions for this already. So, we just call this here `convert` and throw away the wrapper which fetches the conversion from a name. The resulting call for a "simple" conversion would look like this:

```python
import climate_categories as cc

conversion = cc.BURDI.conversion_to(cc.IPCC2006_PRIMAP)
da.convert("category", conversion=conversion)
```

for me, this is actually more transparent than what the current `convert` does:

```python
da.convert("category", "IPCC2006_PRIMAP")
```

Sure, the current `convert` function needs less typing, but it also hides where the conversions come from, so as soon as you define your own categorization which is not built into climate_categories, it is unclear how you would use it. In contrast, using the "inner" function as the public API, it is obvious that you can build your own conversion and pass that.

Reply:
I think I originally designed the API of `convert` to be similar to e.g. GWP and unit conversion. However, for units and also global warming potentials, we assume that user-defined units and GWPs will be super uncommon, so hiding the complexities from the user made sense. So, users don't have to pass a `Unit` object, they can simply pass a string with a name.

However, here, I think we already know from the non-AnnexI repo that we will frequently have to modify the `conversion` before we can use (or use a fully custom conversion). In this case, users have a `Conversion` object already anyway, so it makes sense to use the `Conversion` object instead of names.

But we should check that the passed `Conversion` actually converts from the actual source categorization of the data and refuse to convert FAOstat categories using a conversion from BURDI to IPCC2006.

Reply:
makes all sense to me

Reply:
Thanks for the input Mika and Johannes. Mika if you could have a quick look at `_convert.py`, that would be great. I have now integrated the changes from the latest climate categories release. In the end, only a few lines of code changed.